UI/homepage.py

```diff
@@ -101,28 +235,65 @@ def handle_ui_events(event):
                 draw_filter_search_screen()
             elif event.ui_element == button_recipe:
                 draw_recipe_search_screen()
-            elif back_button and event.ui_element == back_button:
+            elif event.ui_element == back_button:
                 title.show()
                 button_filter.show()
                 button_recipe.show()
                 back_button.hide()
-                input_box.hide()
-                filter_title.hide()
                 if recipe_title:
                     recipe_title.hide()
-                search_results.hide()
+                if recipe_search_button:
+                    recipe_search_button.hide()
+                for checkbox in check_boxes:
+                    checkbox.checked = False
+                for filter_button in filter_buttons:
+                    filter_button.hide()
                 current_screen = 'home'
+
+                if input_box:
```

Comment:
here you're calling basically the same if statement multiple times, so something like this could be simpler:
```
objects_being_checked = [input_box, items_text_box, plus_button] #etc and add the rest + make a better variable name
for object in objects_being_checked:
  if object:
    object.hide()
```